streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordQueue.java

```diff
@@ -189,6 +191,11 @@ private void updateHead() {
             }
 
             headRecord = new StampedRecord(deserialized, timestamp);
+
+            // update the partition timestamp if the current head record's timestamp has exceed its value
```

Comment:
Do we need this comment? Seem redundant.

Reply:
Fair enough